### PR TITLE
v-rmanda/191197/do_select_on_status_and_Id_parameters

### PR DESCRIPTION
**Pull request by v-rmanda**
implemented $Select for submission for status, id parameters

### CODE REVIEW

File: samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs [hunk 1]
```diff
@@ -44,9 +44,13 @@ public static void ValidateConfiguration(IConfiguration configuration)
             {
                 throw new Exception("Missing studentAccount please check appconfig.json file.");
             }
-            else if (string.IsNullOrEmpty(_config["password"]))
+            else if (string.IsNullOrEmpty(_config["teacherpassword"]))
             {
-                throw new Exception("Missing password please check appconfig.json file.");
+                throw new Exception("Missing student password please check appconfig.json file.");
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs]:
```suggestion
                throw new Exception("Missing teacher password please check appconfig.json file.");
```

File: samples/csharp/MicrosoftEduGraphSamples/appsettings.json [hunk 2]
```diff
@@ -5,5 +5,7 @@
   "classId": "YOUR_CLASS_ID",
   "teacherAccount": "YOUR_TEACHER_ACCOUNT",
   "studentAccount": "YOUR_STUDENT_ACCOUNT",
-  "password": "YOUR_PASSWORD"
+  "teacherpassword": "YOUR_PASSWORD",
+  "studentpassword": "YOUR_PASSWORD"
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/appsettings.json]: it seems to be a breaking change. You need to update all workflows and test to make sure they are working fine.

File: samples/csharp/MicrosoftGraphSDK/Submission.cs [hunk 3]
```diff
@@ -96,6 +96,37 @@ public static async Task<EducationSubmissionCollectionResponse> GetSubmissionsWi
...
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Submissions
+                    .GetAsync((requestConfiguration) =>
+                    {
+                        requestConfiguration.QueryParameters.Select = new string[] { "status","id" };
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftGraphSDK/Submission.cs]: you must use the `select` parameter here, not the properties

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs [hunk 4]
```diff
@@ -60,6 +61,18 @@ public async Task ReassignWorkflow()
                     throw new Exception($"No submission found for {_config["studentAccount"]} in {assignmentId} for class {_config["classId"]}");
                 }
 
+                // Get the student submission using Select
+                var submissionsSelect = await Submission.GetSubmissionsWithSelectAsync(graphClientStudentRole, _config["classId"], assignmentId, "status");
+                if (submissionsSelect.Value.Count > 0)
+                {
+                    submissionIdSelect = submissionsSelect.Value[0].Id;
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs:68]: make sure to verify the properties you specified in the `select` are available, here you are only checking the `id` property.

**Review by cristobal-buenrostro — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/appsettings.json (hunk 2 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/appsettings.json]: verified for all workflows

File: samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs (hunk 4 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs:68]: verified for id and status

File: samples/csharp/MicrosoftGraphSDK/Submission.cs (hunk 3 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftGraphSDK/Submission.cs]: updated to select and made dependent changes.

File: samples/csharp/MicrosoftEduGraphSamples/appsettings.json [hunk 5]
```diff
@@ -5,5 +5,7 @@
   "classId": "YOUR_CLASS_ID",
   "teacherAccount": "YOUR_TEACHER_ACCOUNT",
   "studentAccount": "YOUR_STUDENT_ACCOUNT",
-  "password": "YOUR_PASSWORD"
+  "teacherpassword": "YOUR_PASSWORD",
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/appsettings.json]:
```suggestion
  "teacherPassword": "TEACHER_PASSWORD",
```

File: samples/csharp/MicrosoftEduGraphSamples/appsettings.json (hunk 2 above)
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/appsettings.json]:
```suggestion
  "studentpassword": "STUDENT_PASSWORD"
```

File: samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs [hunk 6]
```diff
@@ -44,9 +44,13 @@ public static void ValidateConfiguration(IConfiguration configuration)
             {
                 throw new Exception("Missing studentAccount please check appconfig.json file.");
             }
-            else if (string.IsNullOrEmpty(_config["password"]))
+            else if (string.IsNullOrEmpty(_config["teacherpassword"]))
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs]: check all samples where you are using `teacherpassword`, and use "P" -> `teacherPassword`

File: samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs [hunk 7]
```diff
@@ -44,9 +44,13 @@ public static void ValidateConfiguration(IConfiguration configuration)
...
             {
-                throw new Exception("Missing password please check appconfig.json file.");
+                throw new Exception("Missing teacher password please check appconfig.json file.");
+            }
+            else if (string.IsNullOrEmpty(_config["studentpassword"]))
```
Comment (cristobal-buenrostro) [inline samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs]: `studentPassword` check for all samples

**Review by cristobal-buenrostro — CHANGES_REQUESTED** [review]

File: samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs (hunk 6 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs]: updated teacherpassword to teacherPassword for all instances and checked all methods of all files.

File: samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs (hunk 7 above)
Comment (v-rmanda) [reply samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs]: updated studentpassword to studentPassword for all instances and checked all samples

**Review by cristobal-buenrostro — APPROVED** [review]

**Pull request merged**